components/core/src/LogTypeDictionaryWriter.cpp

```diff
@@ -90,16 +88,17 @@ bool LogTypeDictionaryWriter::add_occurrence (std::unique_ptr<LogTypeDictionaryE
 
         // Insert new entry into dictionary
         auto entry_ptr = entry_wrapper.release();
-        m_value_to_entry[value] = entry_ptr;
-        m_id_to_entry[logtype_id] = entry_ptr;
-
-        // Mark ID as dirty
-        m_uncommitted_entries.emplace_back(entry_ptr);
+        m_value_to_id[value] = logtype_id;
```

Comment:
Same comment as in `VariableDictionaryWriter` except in this case, you'll need to change where `entry_wrapper` is allocated, etc.

Reply:
Fixed this and other related places in the commit fa435ab5f3f74951694c85112e0d36453c7d5347

Reply:
just came up with a better way to create the new function. so you can't ignore the function and wait for my next commit (which might happen tmr morning)